### PR TITLE
Keyboard Interface

### PR DESCRIPTION
**Pull request by SunnevaThorsteins**
Users are now able to navigate WebRICE only using the keyboard in line with [accessibility guidelines](https://www.w3.org/WAI/WCAG21/quickref/?showtechniques=211%2C214#keyboard-accessible). All event listeners are an alternative for a 'click' event. With this, we have no key combination which automatically selects WebRICE. It is unnecessary to be in line with AA accessibility but we may wish to add it for user convenience later. 

Edit: all commits are available now.

### CODE REVIEW

File: src/modules/SpeedButton.ts [hunk 1]
```diff
@@ -11,8 +11,6 @@ export class SpeedButton extends MainButton {
     speedSettings = [0.5, 0.75, 1, 1.25, 1.5, 1.75, 2];
     speedSelectId = 'webriceSelect';
     speedElementClass = 'webriceSpeedElement';
-    // eslint-disable-next-line no-invalid-this
-    hideSpeedModule = this.closeUnusedSpeedComponents.bind(this);
```
Comment (judyfong) [inline src/modules/SpeedButton.ts:15]: removing this and binding directly in the eventlistener reintroduces the run away eventlistener bug again fixed by #29

File: src/modules/SpeedButton.ts [hunk 2]
```diff
@@ -83,7 +81,22 @@ export class SpeedButton extends MainButton {
         (document.getElementById(this.id) as HTMLDivElement)
             .setAttribute('aria-expanded', 'true');
         readingSpeedsElement.style.display = 'block';
-        document.addEventListener('click', this.hideSpeedModule);
+        document.addEventListener('click',
+            this.closeUnusedSpeedComponents.bind(this));
+        document.addEventListener('keydown', (e) => {
+          this.keyboardCloseUnused(e);
```
Comment (judyfong) [inline src/modules/SpeedButton.ts:87]: this should probably be done a different way

**Review by judyfong — CHANGES_REQUESTED** [review]

File: src/modules/SpeedButton.ts [hunk 3]
```diff
@@ -83,7 +81,22 @@ export class SpeedButton extends MainButton {
         (document.getElementById(this.id) as HTMLDivElement)
             .setAttribute('aria-expanded', 'true');
         readingSpeedsElement.style.display = 'block';
-        document.addEventListener('click', this.hideSpeedModule);
+        document.addEventListener('click',
+            this.closeUnusedSpeedComponents.bind(this));
```
Comment (judyfong) [inline src/modules/SpeedButton.ts:85]: this should probably be done a different way

**Comment by judyfong** [comment]
It looks good. There's just the problem of the many eventlisteners.

**Comment by SunnevaThorsteins** [comment]
Fixed the many event listeners bug I recreated so it should be set to merge

File: src/modules/SpeedButton.ts [hunk 4]
```diff
@@ -83,7 +83,22 @@ export class SpeedButton extends MainButton {
         (document.getElementById(this.id) as HTMLDivElement)
             .setAttribute('aria-expanded', 'true');
         readingSpeedsElement.style.display = 'block';
-        document.addEventListener('click', this.hideSpeedModule);
+        document.addEventListener('click',
+            this.closeUnusedSpeedComponents.bind(this.hideSpeedModule));
```
Comment (judyfong) [inline src/modules/SpeedButton.ts:87]:
```suggestion
            this.hideSpeedModule);
```

File: src/modules/SpeedButton.ts [hunk 5]
```diff
@@ -98,7 +113,11 @@ export class SpeedButton extends MainButton {
           .setAttribute('aria-expanded', 'false');
       // Remove document.eventListener for hiding the speedoptions if anywhere
       // other than the speed stuff is clicked
-      document.removeEventListener('click', this.hideSpeedModule);
+      document.removeEventListener('click',
+          this.closeUnusedSpeedComponents.bind(this));
```
Comment (judyfong) [inline src/modules/SpeedButton.ts:117]:
```suggestion
          this.hideSpeedModule);
```

**Review by judyfong — COMMENTED** [review]
I think there was a bit of confusion about the specific variable. So now I've added specific line suggestions.

**Comment by SunnevaThorsteins** [comment]
Suggestions have been committed :)

**Review by judyfong — APPROVED** [review]

**Pull request merged**